### DISCUSSION TITLE
Returning specific status codes for Cloud Events

### DISCUSSION THREAD

**Discussion by jimmyff** [GoogleCloudPlatform/functions-framework-dart#271] [Q&A] [answered]
When using the `cloudevent` function signature I don't get the shelf `Response` object. My function looks like so:

```dart
@CloudFunction()
Future<void> function(PubSub pubSub, RequestContext context) async {
    //...
```

How can I return specific Http status codes? I've tried setting `context.responseHeaders['Status-Code']` to something but that doesn't work.

Thanks

**Comment by kevmoo** [comment]
Thoughts @subfuzion and @grant ?

**Reply by grant** [reply]
https://github.com/GoogleCloudPlatform/functions-framework#http-status-codes

For event-driven functions, we usually keep the interface minimal to cover CloudEvent semantics and not HTTP semantics. The framework will return 200 if nothing goes wrong. 4XX if the function throws an error, etc. You can always use the HTTP signature and inspect headers / return JSON.

Returning nothing or a CloudEvent is usually OK.

**Comment by kevmoo** [comment]
If you throw a random error it will end up sending a 500. And I think there
is an exception type that if you throw it means that it was endowed request
and will return a 400 type. I'm away from my keyboard right now, so I'll
have to look

**Comment by kevmoo** [comment, marked as answer]
Here you go: https://pub.dev/documentation/functions_framework/latest/serve/BadRequestException-class.html

Through this to send a `400` – other exceptions send a `500`

**Reply by kevmoo** [reply]
Ooo...sorry. You can specific the "flavor" of 4xx to send – 400 to 499 😄
